mm/slab.c

```diff
@@ -3481,7 +3495,7 @@ void ___cache_free(struct kmem_cache *cachep, void *objp,
  */
 void *kmem_cache_alloc(struct kmem_cache *cachep, gfp_t flags)
 {
-	void *ret = slab_alloc(cachep, flags, _RET_IP_);
+	void *ret = slab_alloc(cachep, flags, 0, _RET_IP_);
```

Comment:
Could we just use cachep->size? That way we'd be explicit about it, because it'll be confusing to reviewers trying to understand this.

I see that kfence_guarded_alloc still has the 'size = override_size ? ...' so this works, but I think if we can get rid of that hack would be better. I think we don't not know the real size anywhere now, right?

Then, in kfence_guarded_alloc, we can have a KFENCE_WARN or so if the size is 0. Or we could declare it a real error. https://lwn.net/Articles/236920/ seems to indicate it's a stupid thing to do and  slub/slab return ZERO_SIZE_PTR if someone decided to do a kmalloc(0).

Reply:
I think you are right.
There's a check for !cachep in slab_alloc_node(), but it is dead already, as cachep is being dereferenced in the callers of `slab_alloc_node()`.